source/server/server.cc

```diff
@@ -151,18 +152,20 @@ void InstanceImpl::initialize(Options& options,
             restarter_.version());
 
   // Handle configuration that needs to take place prior to the main configuration load.
-  Json::ObjectSharedPtr config_json = Json::Factory::loadFromFile(options.configPath());
   envoy::api::v2::Bootstrap bootstrap;
   if (!options.bootstrapPath().empty()) {
```

Comment:
Aren't we at parity now? Do we need two options? Can we autodetect somehow and just switch back to -c only?

Reply:
How do you propose doing this? Attempt to parse into a proto, if this fails then print a warning and switch to v1 JSON? We will have both v1 and v2 `.json` files.

Reply:
Yeah, that's what I would do. For now, we could not have a warning, but as we move towards v1 deprecation and we have docs, we could start printing a warning that says something like "v1 configuration detected, please switch to v2" etc.